packages/service-library/tests/test_utils.py

```diff
@@ -91,3 +92,45 @@ def print_tree(path: Path, level=0):
     print(f"{tab}{'+' if path.is_dir() else '-'} {path if level==0 else path.name}")
     for p in path.glob("*"):
         print_tree(p, level + 1)
+
+
+@pytest.fixture(params=[Awaitable, Coroutine])
+def future_type(request) -> Type[Union[asyncio.Future, Awaitable]]:
+    return request.param
+
+
+@pytest.fixture()
+async def future_to_test(
+    future_type: Type[Union[asyncio.Future, Awaitable]]
+) -> Union[asyncio.Future, Awaitable]:
+    async def _self_cancelling() -> None:
+        raise asyncio.CancelledError("manual cancellation")
+
+    if future_type == Awaitable:
+        return asyncio.ensure_future(_self_cancelling())
```

Comment:
FYI, https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task recommends using ``asyncio.create_task`` instead of the lowerlevel ``asyncio.ensure_future``.